2.15.0beta4 crash on custom sync server URL bad hostname

Looks like this just wasn't bubbled up to the user? it crashed?

```
com.ichi2.libanki.sync.CustomSyncServerUrlException: customSyncServerUrl
at com.ichi2.libanki.sync.HttpSyncer.parseUrl(HttpSyncer.java:3)
at com.ichi2.libanki.sync.HttpSyncer.req(HttpSyncer.java:33)
at com.ichi2.libanki.sync.HttpSyncer.req(HttpSyncer.java:2)
at com.ichi2.libanki.sync.RemoteServer.hostKey(RemoteServer.java:5)
at com.ichi2.async.Connection.doInBackgroundLogin(Connection.java:5)
at com.ichi2.async.Connection.doOneInBackground(Connection.java:3)
at com.ichi2.async.Connection.doInBackground(Connection.java:4)
at com.ichi2.async.Connection.doInBackground(Connection.java:1)
at android.os.AsyncTask$2.call(AsyncTask.java:333)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.IllegalArgumentException: Expected URL scheme 'http' or 'https' but no colon was found
at okhttp3.HttpUrl$Builder.parse$okhttp(HttpUrl.kt:48)
at okhttp3.HttpUrl$Companion.get(HttpUrl.kt:1)
at okhttp3.HttpUrl.get(Unknown Source:2)
at com.ichi2.libanki.sync.HttpSyncer.parseUrl(HttpSyncer.java:1)
```

--- a/AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java
@@ -39,6 +39,7 @@ import com.google.android.material.textfield.TextInputLayout;
 import com.ichi2.anki.web.HostNumFactory;
 import com.ichi2.async.Connection;
 import com.ichi2.async.Connection.Payload;
+import com.ichi2.libanki.sync.CustomSyncServerUrlException;
 import com.ichi2.themes.StyledProgressDialog;
 import com.ichi2.ui.TextInputEditField;
 import com.ichi2.utils.AdaptionUtil;
@@ -309,6 +310,11 @@ public class MyAccount extends AnkiActivity {
     protected String getHumanReadableLoginErrorMessage(Exception exception) {
         if (exception == null) {
             return "";
+        }
+
+        if (exception instanceof CustomSyncServerUrlException) {
+            String url = ((CustomSyncServerUrlException)exception).getUrl();
+            return getResources().getString(R.string.sync_error_invalid_sync_server, url);
         }
 
         if (exception.getCause() != null) {

--- a/AnkiDroid/src/main/java/com/ichi2/async/Connection.java
+++ b/AnkiDroid/src/main/java/com/ichi2/async/Connection.java
@@ -34,6 +34,7 @@ import com.ichi2.anki.R;
 import com.ichi2.anki.exception.MediaSyncException;
 import com.ichi2.anki.exception.UnknownHttpResponseException;
 import com.ichi2.libanki.Collection;
+import com.ichi2.libanki.sync.CustomSyncServerUrlException;
 import com.ichi2.libanki.sync.FullSyncer;
 import com.ichi2.libanki.sync.HostNum;
 import com.ichi2.libanki.sync.HttpSyncer;
@@ -233,7 +234,13 @@ public class Connection extends BaseAsyncTask<Connection.Payload, Object, Connec
             Timber.w(e);
             data.success = false;
             data.resultType = ERROR;
-            data.result = new Object[]{e.getResponseCode(), e.getMessage() };
+            data.result = new Object[] {e.getResponseCode(), e.getMessage()};
+            return data;
+        } catch (CustomSyncServerUrlException e2) {
+            Timber.w(e2);
+            data.success = false;
+            data.resultType = CUSTOM_SYNC_SERVER_URL;
+            data.result = new Object[] {e2};
             return data;
         } catch (Exception e2) {
             Timber.w(e2);

